tests/integration/long/test_failure_types.py

```diff
@@ -0,0 +1,295 @@
+# Copyright 2013-2015 DataStax, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+try:
+    import unittest2 as unittest
+except ImportError:
+    import unittest
+
+from cassandra.query import (PreparedStatement, BoundStatement, SimpleStatement,
+                             BatchStatement, BatchType)
+from cassandra.cluster import Cluster
+from cassandra import ConsistencyLevel
+from cassandra import WriteFailure, ReadFailure, FunctionFailure
+import time
+from tests.integration import use_singledc, PROTOCOL_VERSION, get_cluster, setup_keyspace
+
+
+def setup_module():
+    """
+    We need some custom setup for this module. All unit tests in this module
+    require protocol >=4. We won't bother going through the setup required unless that is the
+    protocol version we are using.
+    """
+
+    # If we aren't at protocol v 4 or greater don't waste time setting anything up, all tests will be skipped
+    if PROTOCOL_VERSION >= 4:
+        use_singledc(start=False)
+        ccm_cluster = get_cluster()
+        ccm_cluster.stop()
+        config_options = {'tombstone_failure_threshold': 2000, 'tombstone_warn_threshold': 1000}
+        ccm_cluster.set_configuration_options(config_options)
+        ccm_cluster.start(wait_for_binary_proto=True, wait_other_notice=True)
+        # there seems to be some race, with some versions of C* taking longer to
+        # get the auth (and default user) setup. Sleep here to give it a chance
+        time.sleep(10)
+        setup_keyspace()
+
+
+def teardown_module():
+    """
+    The rests of the tests don't need custom tombstones
+    reset the config options so as to not mess with other tests
+    """
+    if PROTOCOL_VERSION >= 4:
+        ccm_cluster = get_cluster()
+        config_options = {}
+        ccm_cluster.set_configuration_options(config_options)
+        if ccm_cluster is not None:
+            ccm_cluster.stop()
+
+
+class ClientExceptionTests(unittest.TestCase):
+
+    def setUp(self):
+        """
+        Test is skipped if run with native protocol version <4
+        """
+
+        if PROTOCOL_VERSION < 4:
```

Comment:
The only reason I don't want to do this is that raising a skiptest exception in the setup module seems to error out rather than mark tests as skipped.